packages/indicators/src/builders/buildArithmetic/buildArithmetic.ts

```diff
@@ -40,14 +40,16 @@ const fetchAnalyticClusters = async (
   return clusters.map(replaceAnalyticValuesWithDefaults).filter(checkClusterIncludesAllElements);
 };
 
-const buildAnalyticValues = (analyticClusters: AnalyticCluster[], formula: string) =>
-  analyticClusters
-    .map(({ organisationUnit, period, dataValues }) => ({
-      organisationUnit,
-      period,
-      value: runArithmetic(formula, dataValues),
-    }))
+const buildAnalyticValues = (analyticClusters: AnalyticCluster[], formula: string) => {
+  const parser = new ExpressionParser();
+
+  return analyticClusters
+    .map(({ organisationUnit, period, dataValues }) => {
+      parser.setScope(dataValues);
```

Comment:
That's a really good observation! I believe it won't cause issues in the current implementations, since we make sure that every cluster includes all data elements (see `checkClusterIncludesAllElements`), so the scope should be complete every time.

But this may change in the future and has a bit of a risk. I will confirm that there is no significant performance toll, and if there is not I'll change it as per your suggestion!

Reply:
Or I could simply use `clearScope`, and problem solved!